cmd/ib-sriov-cni/main.go

```diff
@@ -164,15 +160,82 @@ func cmdAdd(args *skel.CmdArgs) error {
 		}()
 	}
 
-	result := &current.Result{}
-	result.Interfaces = []*current.Interface{{
-		Name:    args.IfName,
-		Sandbox: netns.Path(),
-	}}
-
 	err = sm.SetupVF(netConf, args.IfName, args.ContainerID, netns)
+	if err != nil {
+		return fmt.Errorf("failed to set up pod interface %q from the device %q: %v", args.IfName, netConf.Master, err)
+	}
+	return nil
+}
+
+// Run the IPAM plugin
+func runIPAMplugin(args *skel.CmdArgs, netConf *localtypes.NetConf, netns ns.NetNS, result **current.Result) error {
```

Comment:
Can this method return the IPAM Result struct

then incorporating the current result with it can be done in the main flow.